File: RepoTransBench/target_projects/Java/C#/knightliao_pikaQ/public_tests/AbstractDaoPublicTest.cs
using System;
using Xunit;
using System.Collections.Generic;
using System.Linq;

namespace KnightliaoPikaQ.PublicTests
{
    public abstract class BaseObject<T>
    {
        public abstract T GetId();
        public abstract void SetId(T id);
    }

    public class Match
    {
        public string Column { get; set; }
        public object Value { get; set; }
        public Match(string column, object value) { Column = column; Value = value; }
    }

    public class Modify
    {
        public string Column { get; set; }
        public object Value { get; set; }
        public Modify(string column, object value) { Column = column; Value = value; }
    }

    public class Order
    {
        public string Column { get; }
        public bool IsAsc { get; }
        public Order(string column, bool asc) { Column = column; IsAsc = asc; }
    }

    public class DaoPage
    {
        public int PageNo { get; set; }
        public int PageSize { get; set; }
    }

    public class DaoPageResult<T>
    {
        public List<T> Result { get; set; }
        public int TotalCount { get; set; }
        public DaoPageResult(List<T> result, int total) { Result = result; TotalCount = total; }
        public List<T> GetResult() => Result;
        public int GetTotalCount() => TotalCount;
    }

    public abstract class AbstractDao<TKey, TEntity>
    {
        public virtual Order Order(string col, bool asc) => new Order(col, asc);
        public virtual Match Match(string col, object val) => new Match(col, val);
        public virtual Modify Modify(string col, object val) => new Modify(col, val);

        public virtual List<object> ToList(params object[] items)
        {
            return items == null ? new List<object>() : new List<object>(items);
        }

        public abstract TEntity Get(TKey id);
        public abstract List<TEntity> Find(List<Match> matchList, List<Order> orderList);
        public abstract List<TEntity> Find(List<Match> matchList, List<Order> orderList, int offset, int limit);
        public abstract int Count(List<Match> matchList);

        public virtual object Like(object value) => value;
        public virtual object Between(object left, object right) => new[] { left, right };
        public virtual object GreaterThan(object val) => val;
        public virtual object LessThan(object val) => val;
        public virtual object Express() => new object();
        public virtual object Not(object val) => val;
        public virtual object Incr(object val) => val;
        public DaoPageResult<TEntity> Page2(List<Match> matchList, DaoPage page)
        {
            return new DaoPageResult<TEntity>(Find(matchList, new List<Order>()), Count(matchList));
        }
    }

    public class AbstractDaoPublicTest
    {
        class SimpleEntity : BaseObject<long>
        {
            public long id;
            public SimpleEntity(long id) { this.id = id; }
            public override long GetId() => id;
            public override void SetId(long id) { this.id = id; }
        }

        class SimpleDao : AbstractDao<long, SimpleEntity>
        {
            public override SimpleEntity Get(long id) => new SimpleEntity(id);
            public override List<SimpleEntity> Find(List<Match> matchList, List<Order> orderList)
                => new List<SimpleEntity> { new SimpleEntity(3L), new SimpleEntity(4L) };
            public override List<SimpleEntity> Find(List<Match> matchList, List<Order> orderList, int offset, int limit)
                => new List<SimpleEntity> { new SimpleEntity(3L), new SimpleEntity(4L) };
            public override int Count(List<Match> matchList) => 77;
        }

        [Fact]
        public void TestOrderMatchModify()
        {
            var dao = new SimpleDao();
            var order = dao.Order("otherCol", false);
            Assert.Equal("otherCol", order.Column);
            Assert.False(order.IsAsc);

            var match = dao.Match("bar", 5);
            Assert.Equal("bar", match.Column);
            Assert.Equal(5, match.Value);

            var modify = dao.Modify("baz", 8);
            Assert.Equal("baz", modify.Column);
            Assert.Equal(8, modify.Value);
        }

        [Fact]
        public void TestToList()
        {
            var dao = new SimpleDao();
            var empty = dao.ToList();
            Assert.Equal(0, empty.Count);
            var vals = dao.ToList("test", 42, 7.8, false);
            Assert.Equal(4, vals.Count);
            Assert.Equal(42, vals[1]);
            Assert.Equal(false, vals[3]);
        }

        [Fact]
        public void TestPage2()
        {
            var dao = new SimpleDao();
            var page = new DaoPage { PageNo = 2, PageSize = 10 };
            var result = dao.Page2(new List<Match> { dao.Match("bar", 456) }, page);
            Assert.NotNull(result.GetResult());
            Assert.Equal(77, result.GetTotalCount());
        }

        [Fact]
        public void TestLikeBetweenGreaterLessExpressNotIncr()
        {
            var dao = new SimpleDao();
            Assert.NotNull(dao.Like("otherStr"));
            Assert.NotNull(dao.Between(2, 8));
            Assert.NotNull(dao.GreaterThan(15));
            Assert.NotNull(dao.LessThan(1));
            Assert.NotNull(dao.Express());
            Assert.NotNull(dao.Not("anotherVal"));
            Assert.NotNull(dao.Incr(10));
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/knightliao_pikaQ/public_tests/abstract_dao_public_test.go
package public_tests

import (
	"testing"
)

type SimpleEntity struct {
	ID int64
}

func (e *SimpleEntity) GetID() int64      { return e.ID }
func (e *SimpleEntity) SetID(id int64)    { e.ID = id }
func NewSimpleEntity(id int64) *SimpleEntity { return &SimpleEntity{ID: id} }

type Order struct {
	Column string
	Asc    bool
}
func (o Order) GetColumn() string { return o.Column }
func (o Order) IsAsc() bool       { return o.Asc }

type Match struct {
	Column string
	Value  interface{}
}
func (m Match) GetColumn() string    { return m.Column }
func (m Match) GetValue() interface{} { return m.Value }

type Modify struct {
	Column string
	Value  interface{}
}
func (m Modify) GetColumn() string    { return m.Column }
func (m Modify) GetValue() interface{} { return m.Value }

type DaoPage struct {
	PageNo   int
	PageSize int
}

type DaoPageResult[T any] struct {
	Result     []T
	TotalCount int
}

type AbstractDao struct{}

func (dao *AbstractDao) Order(column string, asc bool) Order {
	return Order{Column: column, Asc: asc}
}
func (dao *AbstractDao) Match(column string, value interface{}) Match {
	return Match{Column: column, Value: value}
}
func (dao *AbstractDao) Modify(column string, value interface{}) Modify {
	return Modify{Column: column, Value: value}
}
func (dao *AbstractDao) ToList(objs ...interface{}) []interface{} {
	return objs
}
func (dao *AbstractDao) Like(_ interface{}) interface{}        { return "like" }
func (dao *AbstractDao) Between(_ interface{}, _2 interface{}) interface{} { return "between" }
func (dao *AbstractDao) GreaterThan(_ interface{}) interface{} { return "greater" }
func (dao *AbstractDao) LessThan(_ interface{}) interface{}    { return "less" }
func (dao *AbstractDao) Express() interface{}                  { return "express" }
func (dao *AbstractDao) Not(_ interface{}) interface{}         { return "not" }
func (dao *AbstractDao) Incr(_ interface{}) interface{}        { return "incr" }

type SimpleDao struct {
	AbstractDao
}

func (sd *SimpleDao) Get(id int64) *SimpleEntity {
	return NewSimpleEntity(id)
}
func (sd *SimpleDao) Find(matchList []Match, orderList []Order) []*SimpleEntity {
	return []*SimpleEntity{NewSimpleEntity(3), NewSimpleEntity(4)}
}
func (sd *SimpleDao) FindPaged(matchList []Match, orderList []Order, offset, limit int) []*SimpleEntity {
	return []*SimpleEntity{NewSimpleEntity(3), NewSimpleEntity(4)}
}
func (sd *SimpleDao) Count(matchList []Match) int { return 77 }
func (sd *SimpleDao) Page2(matchList []Match, page DaoPage) DaoPageResult[*SimpleEntity] {
	return DaoPageResult[*SimpleEntity]{Result: sd.Find(matchList, nil), TotalCount: 77}
}

func TestOrderMatchModify(t *testing.T) {
	dao := &SimpleDao{}
	order := dao.Order("otherCol", false)
	if order.GetColumn() != "otherCol" {
		t.Errorf("order.Column = %v, expected otherCol", order.GetColumn())
	}
	if order.IsAsc() {
		t.Error("order.Asc = true, expected false")
	}
	match := dao.Match("bar", 5)
	if match.GetColumn() != "bar" {
		t.Errorf("match.Column = %v, expected bar", match.GetColumn())
	}
	if match.GetValue() != 5 {
		t.Errorf("match.Value = %v, expected 5", match.GetValue())
	}
	modify := dao.Modify("baz", 8)
	if modify.GetColumn() != "baz" {
		t.Errorf("modify.Column = %v, expected baz", modify.GetColumn())
	}
	if modify.GetValue() != 8 {
		t.Errorf("modify.Value = %v, expected 8", modify.GetValue())
	}
}

func TestToList(t *testing.T) {
	dao := &SimpleDao{}
	empty := dao.ToList()
	if len(empty) != 0 {
		t.Errorf("empty list size = %d, expected 0", len(empty))
	}
	vals := dao.ToList("test", 42, 7.8, false)
	if len(vals) != 4 {
		t.Errorf("vals size = %d, expected 4", len(vals))
	}
	if vals[1] != 42 {
		t.Errorf("vals[1] = %v, expected 42", vals[1])
	}
	if vals[3] != false {
		t.Errorf("vals[3] = %v, expected false", vals[3])
	}
}

func TestPage2(t *testing.T) {
	dao := &SimpleDao{}
	page := DaoPage{PageNo: 2, PageSize: 10}
	result := dao.Page2([]Match{dao.Match("bar", 456)}, page)
	if result.Result == nil {
		t.Error("result.Result == nil")
	}
	if result.TotalCount != 77 {
		t.Errorf("result.TotalCount == %d, expected 77", result.TotalCount)
	}
}

func TestLikeBetweenGreaterLessExpressNotIncr(t *testing.T) {
	dao := &SimpleDao{}
	if dao.Like("otherStr") == nil {
		t.Error("Like(otherStr) == nil")
	}
	if dao.Between(2, 8) == nil {
		t.Error("Between(2,8) == nil")
	}
	if dao.GreaterThan(15) == nil {
		t.Error("GreaterThan(15) == nil")
	}
	if dao.LessThan(1) == nil {
		t.Error("LessThan(1) == nil")
	}
	if dao.Express() == nil {
		t.Error("Express() == nil")
	}
	if dao.Not("anotherVal") == nil {
		t.Error("Not('anotherVal') == nil")
	}
	if dao.Incr(10) == nil {
		t.Error("Incr(10) == nil")
	}
}